extensions/agroal/runtime/src/main/java/io/quarkus/agroal/runtime/AgroalRecorder.java

```diff
@@ -22,8 +21,8 @@ public DataSourceSupport get() {
 
     public Supplier<AgroalDataSource> agroalDataSourceSupplier(String dataSourceName,
             @SuppressWarnings("unused") DataSourcesRuntimeConfig dataSourcesRuntimeConfig) {
-        final AgroalDataSource agroalDataSource = Arc.container().instance(DataSourceProducer.class).get()
-                .createDataSource(dataSourceName);
+        final AgroalDataSource agroalDataSource = Arc.container().instance(DataSources.class).get()
```

Comment:
Now you could use `DataSources.fromName()` here, or?

Reply:
Oops, I forgot to add that :)

Reply:
Fixed